x-pack/plugins/observability/public/pages/overview/loading_observability.tsx

```diff
@@ -21,9 +31,52 @@ const CentralizedFlexGroup = styled(EuiFlexGroup)`
 `;
 
 export function LoadingObservability() {
+  const [allLoaded, setAllLoaded] = useState(false);
   const theme = useContext(ThemeContext);
 
-  return (
+  const { setSharedData } = useContext(ObsvSharedContext);
+
+  const history = useHistory();
+
+  const { absStart, absEnd } = useQueryParams();
+
+  const { data: logs, status: logsStatus } = useInfraLogsHasData();
+  const { data: metrics, status: metricsStatus } = useInfraMetricsHasData();
+  const { data: apm, status: apmStatus } = useApmHasData();
+  const { data: uptime, status: uptimeStatus } = useUptimeHasData();
+  const { data: ux, status: uxStatus } = useUxHasData({ start: absStart, end: absEnd });
+
+  useEffect(() => {
+    const hasAnyData = logs || metrics || apm || uptime || ux;
+
+    if (hasAnyData) {
+      setAllLoaded(true);
+      history.push({ pathname: '/overview' });
+    } else if (
+      logsStatus === 'success' &&
+      metricsStatus === 'success' &&
+      apmStatus === 'success' &&
+      uptimeStatus === 'success' &&
+      uxStatus === 'success'
+    ) {
+      history.push({ pathname: '/landing' });
+      setAllLoaded(true);
+    }
+
+    const hasData = {
+      infra_logs: logs,
+      infra_metrics: metrics,
+      apm,
+      uptime,
+      ux,
+    };
+
+    setSharedData({ hasData, hasAnyData });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [logs, metrics, apm, uptime, ux, history]);
```

Comment:
@shahzad31 thanks for doing it, but I wonder if this is the correct place to add the fetch logic, I don't think `LoadingObservability` should know how to fetch and redirect the user based on the response. 

This also introduces a bug. The user should be able to navigate to the `landing` manually, even if he/she has data. And since you moved the `<LoadingObservability />` component to `application/index.tsx` it no longer possible, the user always gets redirected to the `overview` page.

Reply:
ahh i see, i didn't realise that. i do agree, possibly this component should be dumb, i guess we can move this info into a hook and then push into a context and use from there.